tests/components/blebox/test_cover.py

```diff
@@ -0,0 +1,381 @@
+    STATE_OPEN,
+    STATE_OPENING,
+    SUPPORT_CLOSE,
+    SUPPORT_OPEN,
+    SUPPORT_SET_POSITION,
+    SUPPORT_STOP,
+)
+from homeassistant.const import (
+    ATTR_DEVICE_CLASS,
+    ATTR_SUPPORTED_FEATURES,
+    SERVICE_CLOSE_COVER,
+    SERVICE_OPEN_COVER,
+    SERVICE_SET_COVER_POSITION,
+    SERVICE_STOP_COVER,
+    STATE_UNKNOWN,
+)
+
+from .conftest import async_setup_entity, mock_feature
+
+from tests.async_mock import ANY, AsyncMock, PropertyMock, call, patch
+
+ALL_COVER_FIXTURES = ["gatecontroller", "shutterbox", "gatebox"]
+FIXTURES_SUPPORTING_STOP = ["gatecontroller", "shutterbox"]
+
+
+@pytest.fixture(name="shutterbox")
+def shutterbox_fixture():
+    """Return a shutterBox fixture."""
+    feature = mock_feature(
+        "covers",
+        blebox_uniapi.cover.Cover,
+        unique_id="BleBox-shutterBox-2bee34e750b8-position",
+        full_name="shutterBox-position",
+        device_class="shutter",
+        current=None,
+        state=None,
+        has_stop=True,
+        is_slider=True,
+    )
+    product = feature.product
+    type(product).name = PropertyMock(return_value="My shutter")
+    type(product).model = PropertyMock(return_value="shutterBox")
+    return (feature, "cover.shutterbox_position")
+
+
+@pytest.fixture(name="gatebox")
+def gatebox_fixture():
+    """Return a gateBox fixture."""
+    feature = mock_feature(
+        "covers",
+        blebox_uniapi.cover.Cover,
+        unique_id="BleBox-gateBox-1afe34db9437-position",
+        device_class="gatebox",
+        full_name="gateBox-position",
+        current=None,
+        state=None,
+        has_stop=False,
+        is_slider=False,
+    )
+    product = feature.product
+    type(product).name = PropertyMock(return_value="My gatebox")
+    type(product).model = PropertyMock(return_value="gateBox")
+    return (feature, "cover.gatebox_position")
+
+
+@pytest.fixture(name="gatecontroller")
+def gate_fixture():
+    """Return a gateController fixture."""
+    feature = mock_feature(
+        "covers",
+        blebox_uniapi.cover.Cover,
+        unique_id="BleBox-gateController-2bee34e750b8-position",
+        full_name="gateController-position",
+        device_class="gate",
+        current=None,
+        state=None,
+        has_stop=True,
+        is_slider=True,
+    )
+    product = feature.product
+    type(product).name = PropertyMock(return_value="My gate controller")
+    type(product).model = PropertyMock(return_value="gateController")
+    return (feature, "cover.gatecontroller_position")
+
+
+async def test_init_gatecontroller(gatecontroller, hass, config):
+    """Test gateController default state."""
+
+    _, entity_id = gatecontroller
+    entry = await async_setup_entity(hass, config, entity_id)
+    assert entry.unique_id == "BleBox-gateController-2bee34e750b8-position"
+
+    state = hass.states.get(entity_id)
+    assert state.name == "gateController-position"
+    assert state.attributes[ATTR_DEVICE_CLASS] == DEVICE_CLASS_GATE
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5462445)